# HG changeset patch
# User Rik <rik@octave.org>
# Date 1425743115 28800
#      Sat Mar 07 07:45:15 2015 -0800
# Node ID d9f6260cf20a02a2f39e1ed57da2408288f42182
# Parent  2a992bd191694e973457512350883115142f3a54
Disable __osmesa_print__ tests on non-unix platforms (bug #44338).

* __osmesa_print__.cc: Use isunix() to verify platform before runing
BIST tests.

diff --git a/libinterp/dldfcn/__osmesa_print__.cc b/libinterp/dldfcn/__osmesa_print__.cc
--- a/libinterp/dldfcn/__osmesa_print__.cc
+++ b/libinterp/dldfcn/__osmesa_print__.cc
@@ -253,34 +253,42 @@ instead.\n\
 
   OSMesaDestroyContext (ctx);
 
 #endif
   return retval;
 }
 
 /*
+## FIXME: osmesa does not work correctly on Windows platforms.
+##        This is not critical, since this facility will mostly be used in
+##        the future for generating the images in Octave's own documentation.
+##        For the moment, disable these tests on PC's and Macs.
 %!testif HAVE_OSMESA
-%! h = figure ("visible", "off");
-%! fn = tempname ();
-%! sombrero ();
-%! __osmesa_print__ (h, fn, "svg");
-%! assert (stat (fn).size, 2692270, -0.1);
-%! unlink (fn);
-%! img = __osmesa_print__ (h);
-%! assert (size (img), [get(h, "position")([4, 3]), 3])
-%! ## Use pixel sum per RGB channel as fingerprint
-%! img_fp = squeeze (sum (sum (img), 2));
-%! assert (img_fp, [52942515; 54167797; 56158178], -0.05);
+%! if (isunix ())
+%!   h = figure ("visible", "off");
+%!   fn = tempname ();
+%!   sombrero ();
+%!   __osmesa_print__ (h, fn, "svg");
+%!   assert (stat (fn).size, 2692270, -0.1);
+%!   unlink (fn);
+%!   img = __osmesa_print__ (h);
+%!   assert (size (img), [get(h, "position")([4, 3]), 3])
+%!   ## Use pixel sum per RGB channel as fingerprint
+%!   img_fp = squeeze (sum (sum (img), 2));
+%!   assert (img_fp, [52942515; 54167797; 56158178], -0.05);
+%! endif
 
 %!testif HAVE_OSMESA
-%! h = figure ("visible", "off");
-%! fn = tempname ();
-%! plot (sin (0:0.1:2*pi));
-%! __osmesa_print__ (h, fn, "svgis2d");
-%! assert (stat (fn).size, 7438, -0.05);
-%! unlink (fn);
-%! img = __osmesa_print__ (h);
-%! assert (size (img), [get(h, "position")([4, 3]), 3])
-%! ## Use pixel sum per RGB channel as fingerprint
-%! img_fp = squeeze (sum (sum (img), 2));
-%! assert (img_fp, [59281711; 59281711; 59482179], -0.05);
+%! if (isunix ())
+%!   h = figure ("visible", "off");
+%!   fn = tempname ();
+%!   plot (sin (0:0.1:2*pi));
+%!   __osmesa_print__ (h, fn, "svgis2d");
+%!   assert (stat (fn).size, 7438, -0.05);
+%!   unlink (fn);
+%!   img = __osmesa_print__ (h);
+%!   assert (size (img), [get(h, "position")([4, 3]), 3])
+%!   ## Use pixel sum per RGB channel as fingerprint
+%!   img_fp = squeeze (sum (sum (img), 2));
+%!   assert (img_fp, [59281711; 59281711; 59482179], -0.05);
+%! endif
 */
